scripts/plotting/tape_recorder.py

```diff
@@ -0,0 +1,543 @@
+
+    if (cmap == "blue2red"):
+        mymap = blue2red_cmap(nlevs, nowhite)
+
+    if (cmap == "precip"):
+        mymap = precip_cmap(nlevs, nowhite)
+
+    if (cmap == "precip_nowhite"):
+        mymap = precip_cmap_nowhite(nlevs)
+
+    if (cmap == 'red2blue'):
+        mymap = red2blue_cmap(nlevs, nowhite)
+
+    clevplot=clevs
+    if (posneg == "pos"):
+        clevplot = clevs[clevs >= 0]
+    if (posneg == "neg"):
+        clevplot = clevs[clevs <= 0]
+
+    ax = fig.add_axes([x1, y1, x2-x1, y2-y1])
+    norm = mpl.colors.Normalize(vmin=cmin, vmax=cmax)
+    
+    if (ticks):
+        clb = mpl.colorbar.ColorbarBase(ax, cmap=mymap,
+           orientation=orient, norm=norm, values=clevplot, ticks=ticks)
+    else:
+        clb = mpl.colorbar.ColorbarBase(ax, cmap=mymap, 
+           orientation=orient, norm=norm, values=clevplot)
+
+    clb.ax.tick_params(labelsize=fsize)
+    clb.set_label(titlestr, fontsize=fsize+2)
+
+    if (contourlines):
+        clevlines = clevs*contourlinescale
+        clevlines = clevlines[np.abs(clevlines) > ci/2.]
+        if (orient=='horizontal'):
+            ax.vlines(clevlines[clevlines > 0],-5,5, colors='black', linestyle='solid')
+            ax.vlines(clevlines[clevlines < 0],-5,5, colors='black', linestyle='dashed')
+        if (orient=='vertical'):
+            ax.hlines(clevlines[clevlines > 0],-10,15, colors='black', linestyle='solid')
+            ax.hlines(clevlines[clevlines < 0],-10,15, colors='black', linestyle='dashed')
+
+    return ax
+
+
+def cosweightlat(darray, lat1, lat2):
+    """Calculate the weighted average for an [:,lat] array over the region
+    lat1 to lat2
+    """
+
+    # flip latitudes if they are decreasing
+    if (darray.lat[0] > darray.lat[darray.lat.size -1]):
+        print("flipping latitudes")
+        darray = darray.sortby('lat')
+
+    region = darray.sel(lat=slice(lat1, lat2))
+    weights=np.cos(np.deg2rad(region.lat))
+    regionw = region.weighted(weights)
+    regionm = regionw.mean("lat")
+
+    return regionm
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+
+def plot_pre_mon(fig, data, pre, ci, cmin, cmax, expname, x1=None, x2=None, y1=None, y2=None, 
+                 oplot=False, ax=None, cmap='precip', taxis='time', paxis='lev', climo_yrs=None):
+    """
+    Plot seasonal cycle, pressure versus time.
+    """
+
+    # move the time axis to the first
+    if (data.dims[1] != taxis):
+        data = data.transpose(..., taxis)
+
+    nlevs = (cmax - cmin)/ci + 1
+    clevs = np.arange(cmin, cmax+ci, ci)
+
+    if (cmap == "blue2red"):
+        mymap = blue2red_cmap(nlevs)
+
+    if (cmap == "precip"):
+        mymap = precip_cmap(nlevs)
+
+    if (cmap == "precip_nowhite"):
+        mymap = precip_cmap_nowhite(nlevs)
+
+    # if overplotting, check for axis input
+    if (oplot and (not ax)):
+        print("This isn't going to work.  If overplotting, specify axis")
+        sys.exit()
```

Comment:
I would replace this with a `return` statement, as otherwise this will kill the entire ADF run (when we really just want the `tape_recorder` plots to be skipped).

Reply:
Honestly, this was in the code I don't know exactly what it does. I'm guessing it has to do with the way the plotting scheme handles at most 25 plots (5x5)?

Changed to a `return` call